### ISSUE TITLE
How to deploy a xitrum application

### ISSUE THREAD

**Issue by dolphineor** [xitrum-framework/xitrum#577]
Hi @ngocdaothanh , I'm a beginner for xitrum. After I glanced over the xitrum demos, I have a question about deploying a xitrum application, using a jar or a war or something else? Could you provide an example? Thanks in advance.

**Comment by ngocdaothanh** [comment]
Normally, to deploy Xitrum to run it in production mode, you run Xitrum as a standalone Java program. That is, to start the Java program, you run the command:
```
java mypackage.MyMainClass
```

For details, please see the Xitrum Guide:
http://xitrum-framework.github.io/guide/3.25/html/en/deploy.html

Quick example using [xitrum-new](https://github.com/xitrum-framework/xitrum-new):

Clone the project:
```
git clone https://github.com/xitrum-framework/xitrum-new.git
```

Compile the project, package it to a JAR file, and gather all dependency JAR files to a single directory:
```
cd xitrum-new
sbt/sbt xitrum-package
```

The directory `target/xitrum` will be created. Deploying Xitrum application means copying this directory to your production server and run it.

There's a utility script to let you start the standalone Java program:
```
cd target/xitrum
script/runner quickstart.Boot
```

* If you see inside the `script/runner` script, you will see the `java` command. There, you can tune JVM memory settings etc.
* `quickstart.Boot` is the main class in the `xitrum-new` project. You can see its [source code](https://github.com/xitrum-framework/xitrum-new/blob/master/src/main/scala/quickstart/Boot.scala).

In short, there's nothing "magic". A Xitrum application is just a normal standalone Java program.

**Comment by dolphineor** [comment]
Thanks very much, since the xitrum based on netty and akka, I'll follow this project with interest all the time and hope that xitrum will be better.
